### PR TITLE
Fixes #12650: Directive  tree filter is very slow

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/12650

### CODE REVIEW

File: rudder-web/src/main/webapp/javascript/rudder/angular/filters.js [hunk 1]
```diff
@@ -200,8 +200,13 @@ app.controller('filterTagCtrl', function ($scope, $http, $location, $timeout, $r
     return !$scope.only.key && !$scope.only.value;
   }
 
+  // Add a grace periot before effectively do the search
```
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/angular/filters.js]: period :)

File: rudder-web/src/main/webapp/javascript/rudder/angular/filters.js [hunk 2]
```diff
@@ -200,8 +200,13 @@ app.controller('filterTagCtrl', function ($scope, $http, $location, $timeout, $r
...
+  var to = null;
```
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/angular/filters.js]: you should use undefined 

by declaring only

> var to;

File: rudder-web/src/main/webapp/javascript/rudder/angular/filters.js [hunk 3]
```diff
@@ -200,8 +200,13 @@ app.controller('filterTagCtrl', function ($scope, $http, $location, $timeout, $r
...
   $scope.searchTree = function(treeId) {
-    $(treeId).jstree('searchtag', $scope.searchStr, $scope.tags, $scope.only);
+    if (to) { clearTimeout(to); }
```
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/angular/filters.js]: and here check that

> if to === undefined

Comment (VinceMacBuche) [reply rudder-web/src/main/webapp/javascript/rudder/angular/filters.js]: In fact you can remove the check, calling clearTimeout on undefined cause no problem

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

File: rudder-web/src/main/webapp/javascript/rudder/angular/filters.js [hunk 4]
```diff
@@ -200,8 +200,13 @@ app.controller('filterTagCtrl', function ($scope, $http, $location, $timeout, $r
     return !$scope.only.key && !$scope.only.value;
   }
 
+  // Add a grace period before effectively do the search
+  var to;
   $scope.searchTree = function(treeId) {
-    $(treeId).jstree('searchtag', $scope.searchStr, $scope.tags, $scope.only);
+    if (typeof to !== 'undefined') { clearTimeout(to); }
```
Comment (VinceMacBuche) [inline rudder-web/src/main/webapp/javascript/rudder/angular/filters.js]: we do not use it and i clearly think you should use 

> to !== undefined

**Comment by VinceMacBuche** [comment]
OK, merging this PR

**Pull request merged**